### PR TITLE
Simplify docs TOC, earthfile reference ordering and other tweaks

### PR DESCRIPTION
**Pull request by vladaionescu**
* Made the ToC not be all expanded by default. This partially reverts the ToC change from last month.
* Reordered the commands on the Earthfile reference page to make it a bit more consistent and also to move experimental stuff towards the bottom of the page
* Other minor tweaks

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -3,63 +3,52 @@
...
+    * [Go](examples/go.md)
+    * [Java](examples/java.md)
+    * [JS](examples/js.md)
+    * [C++](examples/cpp.md)
+    * [Python](examples/python.md)
+* Other Earthfile examples
+    * [Monorepo](examples/monorepo.md)
+    * [Multirepo](examples/multirepo.md)
+    * [Earthly](examples/earthly.md)
+* [CI examples](ci-examples/ci-examples.md)
```
Comment (alexcb) [inline docs/SUMMARY.md:51]: This point to `docs/ci-examples/ci-examples.md`, which seems to have been deleted in this PR; or am I missing something here?

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -3,63 +3,52 @@
...
+    * [Cloud secrets](guides/cloud-secrets.md)
+    * [Integration Testing](guides/integration.md)
+    * [Debugging techniques](guides/debugging.md)
+    * [Multi-platform builds](guides/multi-platform.md)
+    * Configuring registries
+        * [AWS ECR](guides/registries/aws-ecr.md)
+        * [GCP Artifact Registry](guides/registries/gcp-artifact-registry.md)
+        * [Azure ACR](guides/registries/azure-acr.md)
+        * [Self-signed certificates](guides/registries/self-signed.md)
 * [Earthfile reference](earthfile/earthfile.md)
```
Comment (alexcb) [inline docs/SUMMARY.md:27]: it looks like `earthfile/earthfile.md` was deleted in this PR, now this links to a broken page?

File: docs/SUMMARY.md (hunk 1 above)
Comment (vladaionescu) [reply docs/SUMMARY.md:51]: Ah forgot to remove the link. Thanks for finding this.

File: docs/SUMMARY.md (hunk 2 above)
Comment (vladaionescu) [reply docs/SUMMARY.md:27]: Hmm no it wasn't.. here's the page in preview: https://app.gitbook.com/@earthly/s/earthly-staging/~/revisions/-MXEhZYiuUi6qVdZamDG/v/vlad%2Fdocs-toc/docs/earthfile

Comment (alexcb) [reply docs/SUMMARY.md:27]: well that's odd; I have no idea why I'm seeing this under the PR:

image

... sigh that's github being github and not showing the change. Sorry my mind just registered it as being deleted. I'll take a look at it.

Comment (vladaionescu) [reply docs/SUMMARY.md:27]: It's almost all just reordering of the commands.

**Review by alexcb — APPROVED** [review]

**Pull request merged**